t/sys/write-read-hole.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2019, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ *
+ * Copyright 2018, UT-Battelle, LLC.
+ *
+ * LLNL-CODE-741539
+ * All rights reserved.
+ *
+ * This is the license for UnifyFS.
+ * For details, see https://github.com/LLNL/UnifyFS.
+ * Please read https://github.com/LLNL/UnifyFS/LICENSE for full license text.
+ */
+
+ /*
+  * Test reading from file with holes
+  */
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include "t/lib/tap.h"
+#include "t/lib/testutil.h"
+
+/* Get global, local, or log sizes (or all) */
+static
+void get_size(char* path, size_t* global, size_t* local, size_t* log)
```

Comment:
Looks like we're getting multiple tests (and will probably get a lot more) that will need this function. We could move this function to [t/lib/testutil.h](https://github.com/LLNL/UnifyFS/blob/dev/t/lib/testutil.h), or could split it into three sharness tests (file_global_size, file_local_size, and file_log_size) and add them to our custom sharness tests in [t/sharness.d/02-functions.sh](https://github.com/LLNL/UnifyFS/blob/dev/t/sharness.d/02-functions.sh).

If either of these sounds good, I can create an issue and change this in a future PR.

Reply:
That makes sense to me.

Reply:
Adam, does this new "Nread" field follow the same logic of the 'nbytes' field that you had suggested in the past? I view it as a kind of water-mark tracking where valid data exists, is this right?

Reply:
Yep, I could have named this nbytes instead -- maybe should have.  I'll get this merged now so you can work from it for HDF testing.  We still need to get the truncate PR in there.